libs/cardano-ledger-core/src/Cardano/Ledger/Keys.hs

```diff
@@ -188,23 +188,8 @@ instance
   toCBOR (VKey vk) = encodeVerKeyDSIGN vk
   encodedSizeExpr _size proxy = encodedVerKeyDSIGNSizeExpr ((\(VKey k) -> k) <$> proxy)
 
--- | Pair of signing key and verification key, with a usage role.
-data KeyPair (kd :: KeyRole) c = KeyPair
-  { vKey :: !(VKey kd c),
-    sKey :: !(DSIGN.SignKeyDSIGN (DSIGN c))
-  }
-  deriving (Generic, Show)
-
-instance
-  ( Crypto c,
-    NFData (DSIGN.VerKeyDSIGN (DSIGN c)),
-    NFData (DSIGN.SignKeyDSIGN (DSIGN c))
-  ) =>
-  NFData (KeyPair kd c)
-
-instance Crypto c => NoThunks (KeyPair kd c)
-
-instance HasKeyRole KeyPair
+newtype KeyPair c = KeyPair ()
+{-# DEPRECATED KeyPair "Use `Test.Cardano.Ledger.Core.KeyPair (KeyPair)` instead" #-}
```

Comment:
@JaredCorduan  - something like this maybe?  I mean leaving it here where the original type was but deprecated, or defining it somewhere else ?

Reply:
yea, exactly, except that I think we want the type to remain the same (ie keep the `KeyRole` and the two fields). The idea is that we want downstream code to still compile, just emit a warning (I don't think that the warning will fire (or be as obvious) if it doesn't type check, but we could test that out).

Reply:
Yeah fair enough, at first I left it as you say but then I wanted to get rid of the imports. But you're right, if we're gonna deprecate it, it's stupid to change the type and leave the name.

Reply:
I am still learning too!